grails-app/domain/org/pih/warehouse/inventory/CycleCountItem.groovy

```diff
@@ -79,4 +79,13 @@ class CycleCountItem {
         }
         return null
     }
+
+    @Override
+    int compareTo(Object that) {
+        int diff = inventoryItem?.expirationDate <=> that.inventoryItem?.expirationDate ?: inventoryItem?.lotNumber <=> that.inventoryItem?.lotNumber
```

Comment:
Currently, on the stock card (on the current stock tab), there is sorting by expiration date and bin location.